ui/appui-react/src/appui-react/messages/StatusMessageRenderer.tsx

```diff
@@ -26,24 +24,34 @@ export interface StatusMessageRendererProps extends CommonProps {
 /** Message Popup React component that renders one or more Toast or Sticky messages and an Activity message without a StatusBar.
  * @note This component was formerly named MessageRenderer in previous releases.
  * @public
+ * @deprecated Use `toaster` from iTwinui-react to display status messages.
  */
 export function StatusMessageRenderer(props: StatusMessageRendererProps) {
-  const [messages, setMessages] = React.useState<ReadonlyArray<StatusMessage>>(MessageManager.activeMessageManager.messages);
+  const messages = React.useRef<{close: () => void}[]>([]);
   const [activityMessageInfo, setActivityMessageInfo] = React.useState<ActivityMessageEventArgs | undefined>(undefined);
-  const [isActivityMessageVisible, setIsActivityMessageVisible] = React.useState(false);
+  const lastToastId = React.useRef(0);
 
   React.useEffect(() => {
-    const handleMessageAddedEvent = (_args: MessageAddedEventArgs) => {
-      setMessages(MessageManager.activeMessageManager.messages);
+    const handleMessageAddedEvent = ({ message }: MessageAddedEventArgs) => {
+      const displayedMessage = MessageManager.displayMessage(
+        message,
+        { onRemove: () => props.closeMessage?.(lastToastId.current.toString()) },
+        { placement: "top", order: "descending" }
+      );
+      if(!!displayedMessage) {
+        messages.current.push(displayedMessage);
+        ++lastToastId.current;
+      }
     };
 
     return MessageManager.onMessageAddedEvent.addListener(handleMessageAddedEvent);
-  }, []);
+  }, [props]);
```

Comment:
List dependencies explicitly, since `props` is a new object on each re-render?

Reply:
Actually good catch. Here I should have empty deps list, because I dont want to add new event listener every time prop changes.